windows/wintools_complete_install_verify/set_vmtools_source.yml

```diff
@@ -0,0 +1,41 @@
+# Copyright 2024 VMware, Inc.
+# SPDX-License-Identifier: BSD-2-Clause
+---
+- name: "Check VMware Tools source parameters"
+  ansible.builtin.assert:
+    that:
+      - (vmtools_url_path is defined and vmtools_url_path) or (vmtools_iso_path is defined and vmtools_iso_path)
+    fail_msg: "One of parameters 'vmtools_url_path', 'vmtools_iso_path' must be set when 'vmtools_esxi_bundled' is set to '{{ vmtools_esxi_bundled }}'."
+  when:
+    - vmtools_esxi_bundled is defined
+    - not vmtools_esxi_bundled | bool
+
+- name: "Set install ESXi bundled VMware Tools by default"
+  ansible.builtin.set_fact:
+    vmtools_esxi_bundled: true
+  when:
+    - vmtools_esxi_bundled is undefined
+    - vmtools_url_path is undefined or not vmtools_url_path
+    - vmtools_iso_path is undefined or not vmtools_iso_path
+
+ - name: "Set install VMware Tools from URL or datastore ISO file"
+   ansible.builtin.set_fact:
+     vmtools_esxi_bundled: false
+   when: vmtools_esxi_bundled is undefined
+
+- name: "Set ESXi bundled VMware Tools ISO path"
+  ansible.builtin.set_fact:
+    vmtools_iso_path: |-
+      {%- if guest_os_ansible_architecture == '32-bit' and esxi_version is version('8.0.3', '>') -%}
+      "[] /vmimages/tools-isoimages/windows-x86.iso"
+      {%- else -%}
+      "[] /vmimages/tools-isoimages/windows.iso"
+      {%- endif -%}
+  when: vmtools_esxi_bundled | bool
```

Comment:
There is a file `/vmimages/tools-isoimages/isoimages_manifest.txt` which provides the mapped windows tools iso file for each VMX guest id. Here shall based on the map to get bundled tools iso. That's closer to user scenario.
```
$ grep windows9 /vmimages/tools-isoimages/isoimages_manifest.txt
windows9 = "windows-x86.iso"
windows9-64 = "windows.iso"
windows9srv-64 = "windows.iso"
```

Reply:
Yeah, that's better. While as we discussed offline, the GOS supported in this test case to install VMware Tools are use 'windows.iso' basically, and getting this file `/vmimages/tools-isoimages/isoimages_manifest.txt` or execute command from/on ESXi server directly is what we are trying to avoid, so we can leave it as it is. Thanks.